nengo/builder/connection.py

```diff
@@ -254,11 +264,23 @@ def get_prepost_signal(is_pre):
         in_signal = slice_signal(model, in_signal, conn.pre_slice)
 
     # Build transform
-    weighted, weights = model.build(conn.transform,
-                                    in_signal,
-                                    decoders=decoders,
-                                    encoders=encoders,
-                                    rng=rng)
+    if conn.solver.weights and not conn.solver.compositional:
+        # special case for non-compositional weight solvers, where
+        # the solver is solving for the full weight matrix, which already
+        # includes the transform
+        weights = Signal(decoders, name="%s.weights" % conn)
```

Comment:
Would it be cleaner to make a new `Dense` transform here and then call the transform builder on that?

Reply:
Done, definitely way cleaner